src/hooks/use-swr-collection.ts

```diff
@@ -521,12 +531,16 @@ export const useCollection = <
 
       docsToAdd.forEach(({ id, ...doc }) => {
         // take the ID out of the document
-        batch.set(ref.doc(id), doc)
+        let docRef = ref.doc(id)
+        if (documentDataConverter) {
+          docRef = docRef.withConverter(documentDataConverter)
```

Comment:
I'm actually not that familiar with how the Firestore converter works. Are you supposed to do this when you upload items too? From the docs, I only see this feature being implemented when it's data being fetched for display to users. https://firebase.google.com/docs/reference/js/firebase.firestore.FirestoreDataConverter

Reply:
Yes, you can use with both reading and writing data, this is why it has both `toFirestore` and `fromFirestore` functions.